data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchAccessor.java

```diff
@@ -5,20 +5,37 @@
 package org.opensearch.dataprepper.plugins.source.opensearch.worker.client;
 
 import org.opensearch.client.opensearch.OpenSearchClient;
-import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.CreatePitRequest;
-import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.CreatePitResponse;
+import org.opensearch.client.opensearch._types.OpenSearchException;
+import org.opensearch.client.opensearch._types.Time;
+import org.opensearch.client.opensearch.core.pit.CreatePitRequest;
+import org.opensearch.client.opensearch.core.pit.CreatePitResponse;
+import org.opensearch.client.opensearch.core.pit.DeletePitRequest;
+import org.opensearch.client.opensearch.core.pit.DeletePitResponse;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.exceptions.SearchContextLimitException;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.CreatePointInTimeRequest;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.CreatePointInTimeResponse;
 import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.CreateScrollRequest;
 import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.CreateScrollResponse;
-import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.DeletePitRequest;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.DeletePointInTimeRequest;
 import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.DeleteScrollRequest;
 import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.SearchContextType;
 import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.SearchPitRequest;
 import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.SearchPitResponse;
 import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.SearchScrollRequest;
 import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.model.SearchScrollResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Objects;
 
 public class OpenSearchAccessor implements SearchAccessor, ClusterClientFactory {
 
+    private static final Logger LOG = LoggerFactory.getLogger(OpenSearchAccessor.class);
+
+    static final String PIT_RESOURCE_LIMIT_ERROR_TYPE = "rejected_execution_exception";
```

Comment:
This error type seems more generic than the PIT logic. Are there other cases where we can get a RejectedExecutionException unrelated to PIT?

Reply:
Probably but what is the alternative? Check the message contains something unique to the PIT message? Even for other exception we would log it and then give up the index and wait to start processing again, which is ok